step-templates/azure-check-vmss-provision-status.json

```diff
@@ -0,0 +1,138 @@
, leaving it alone.\"\n        \n        $addToOutputArray = $true\n        if ($excludeOldServers.ToLower().Trim() -eq \"yes\")\n        {\n        \tWrite-Host \"Pulling back the creation event for $($deploymentTarget.Name)\"\n        \t$creationTasks = Invoke-OctopusApi -octopusUrl $octopusUrl -endPoint \"events?regarding=$($deploymentTarget.Id)&spaces=$($octopusSpaceId)&includeSystem=true&eventCategories=Created\" -spaceId $null -apiKey $apiKey -method \"GET\"\n            \n            $creationDate = @($creationTasks.Items)[0].Occurred\n            Write-Host \"The deployment target $($deploymentTarget.Name) was created on $creationDate\"\n            $differenceInCreationTime = [DateTime]$currentDeploymentQueued - [DateTime]$creationDate\n            \n            Write-Host \"The difference in minutes between creation date and current task queued time is $($differenceInCreationTime.TotalMinutes) minutes\" \n            if ($differenceInCreationTime.TotalMinutes -gt 3)\n            {\n            \tWrite-Host \"The deployment target $($deploymentTarget.Name) existed for more than 3 minutes before this was ran and the excludeOldServers was set to yes, removing this from the output\"\n            \t$addToOutputArray = $false\n            }\n        }\n        \n        if ($addToOutputArray -eq $true)\n        {\n\t        $octopusDeployTargetIds += $deploymentTarget.Id\n    \t    $octopusDeployTargetNames += $deploymentTarget.Name\n        }\n    }\n}\n\nWrite-Highlight \"The Azure VM Scale Set $vmssScaleSetName and Octopus Deploy target list have been successfully reconciled.\"\n\nWrite-Highlight \"Setting the output variable 'VMSSDeploymentTargetIds' to $($octopusDeployTargetIds -join \",\").  Please note, any old servers were removed from this list if exclude old servers is set to yes.\"\nSet-OctopusVariable -Name \"VMSSDeploymentTargetIds\" -Value ($octopusDeployTargetIds -join \",\")\n\nWrite-Highlight \"Setting the output variable 'VMSSDeploymentTargetNames' to $($octopusDeployTargetNames -join \",\"). Please note, any old servers were removed from this list if exclude old servers is set to yes\"\nSet-OctopusVariable -Name \"VMSSDeploymentTargetNames\" -Value ($octopusDeployTargetNames -join \",\")\n\n$vmssHasServers = $octopusDeployTargetIds.Count -gt 0\nif ($duplicateRun -eq $true)\n{\n\tWrite-Highlight \"Duplicate run detected, therefore there are no new servers to deploy to.  Setting the output variable 'VMSSHasServers' to $false\"\n    $vmssHasServers = $false\n}\nelseif ($excludeOldServers.ToLower().Trim() -eq \"yes\" -and $vmssHasServers -eq $false)\n{\n\tWrite-Highlight \"Setting the output variable 'VMSSHasServers' to $vmssHasServers.  Exclude old server was set to 'Yes' and there are no servers.  This most likely means there were no new servers found.\"\n}\nelse\n{\n\tWrite-Highlight \"Setting the output variable 'VMSSHasServers' to $vmssHasServers.\"\n}\nSet-OctopusVariable -Name \"VMSSHasServers\" -Value $vmssHasServers\n"
```

Comment:
One minor comment on the main script body:
- On or around L207, the script outputs `The VMSS is still provisioning, sleeping for 10 seconds then checking again.` but the sleep is for 5 seconds e.g. `Start-Sleep -Seconds 5`